crates/miden-lib/src/account/auth/mod.rs

```diff
@@ -39,3 +42,63 @@ impl From<RpoFalcon512> for AccountComponent {
         .with_supports_all_types()
     }
 }
+
+/// An [`AccountComponent`] implementing a conditional RpoFalcon512 signature scheme for
+/// authentication of transactions.
+///
+/// This component only requires authentication when any of the specified procedures are called
+/// during the transaction. It stores a list of procedure roots that require authentication, and
+/// the signature verification is only performed if at least one of these procedures was invoked.
+///
+/// It exports the procedure `auth__tx_rpo_falcon512_conditional`, which:
+/// - Checks if any of the specified procedures were called during the transaction
+/// - If none were called, authentication is skipped
+/// - If at least one was called, performs standard RpoFalcon512 signature verification
+///
+/// The storage layout is:
+/// - Slot 0: Public key (same as RpoFalcon512)
+/// - Slot 1: Number of tracked procedures
+/// - Slots 2+: Procedure roots that trigger authentication (one Word per procedure)
+///
+/// This component supports all account types.
+pub struct RpoFalcon512Conditional {
```

Comment:
nit: I don't really love the `RpoFalcon512Conditional` name. Maybe `RpoFalcon512Ext` would be better, though it is not great either.

Reply:
I like `RpoFalcon512ProcedureGated` and `RpoFalcon512ProcedureACL`, they're the most very descriptive. I agree ACL should be clear, and would make `RpoFalcon512ProcedureACL` my preference.